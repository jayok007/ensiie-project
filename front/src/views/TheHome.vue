<template>
  <div>
    <v-card color="transparent" flat>
        <v-card-text class="light-green--text">
          <h1 class="text-xs-center">La flemme de chercher un bar ?</h1>
          <h1 class="text-xs-center">On s'occupe de tout !</h1>
        </v-card-text>
    </v-card>

    <v-layout row wrap>
      <v-flex xs12 lg8 xl6 offset-lg2 offset-xl3>
        <search-bar
          :keywords="keywords"
          v-model="selectedKeywords"
          @search="search"
        ></search-bar>
      </v-flex>
    </v-layout>
  </div>
</template>

<script>
import SearchBar from '@/components/SearchBar'

export default {
  name: 'Home',

  components: {
    SearchBar
  },

  data () {
    return {
      keywords: [],
      selectedKeywords: []
    }
  },

  created () {
    this.$api.getKeywords()
      .then(keywords => (this.keywords = keywords))
      .catch(this.$log.error)
  },

  methods: {
    search () {
      if (this.selectedKeywords.length === 0) return

      this.$router.push(`/search?q=${this.selectedKeywords.join(',')}`)
    }
  }
}
</script>
