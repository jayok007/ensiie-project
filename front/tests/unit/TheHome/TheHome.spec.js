import { shallowMount } from '@vue/test-utils'
import TheHome from '@/views/TheHome'

describe('TheHome', () => {
  it('should render properly', () => {
    const wrapper = shallowMount(TheHome, {
      mocks: {
        $store: {
          dispatch: jest.fn(),
          state: {
            keywords: ['a', 'b', 'c', 'd', 'e']
          }
        }
      },
      data () {
        return {
          selectedKeywords: ['a']
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
