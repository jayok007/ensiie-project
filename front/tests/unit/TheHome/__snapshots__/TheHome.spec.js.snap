// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHome should render properly 1`] = `
<vcontent-stub tag="main">
  <vcontainer-stub tag="div" fluid="">
    <vlayout-stub tag="div" align-center="" justify-center="" row="" wrap="">
      <vflex-stub tag="div" xs12="">
        <vcard-stub flat="true" tag="div" color="transparent">
          <vcardtext-stub class="purple-color">
            <h1 class="text-xs-center">La flemme de chercher un bar ?</h1>
            <h1 class="text-xs-center">On s'occupe de tout !</h1>
          </vcardtext-stub>
        </vcard-stub>
      </vflex-stub>
      <vflex-stub tag="div" xs12="" lg8="" xl6="">
        <searchbar-stub keywords="a,b,c,d,e" value="a" class="search-bar"></searchbar-stub>
      </vflex-stub>
    </vlayout-stub>
  </vcontainer-stub>
</vcontent-stub>
`;
